cdc/sink/flowcontrol/flow_control.go

```diff
@@ -72,45 +88,53 @@ func NewTableFlowController(quota uint64) *TableFlowController {
 func (c *TableFlowController) Consume(
 	msg *model.PolymorphicEvent,
 	size uint64,
-	callBack func(batch bool) error,
+	callBack func(batchID uint64) error,
 ) error {
 	commitTs := msg.CRTs
 	lastCommitTs := atomic.LoadUint64(&c.lastCommitTs)
+	blockingCallBack := func() error {
+		err := callBack(c.batchID)
+
+		if commitTs == lastCommitTs {
+			c.batchID++
```

Comment:
Good idea!

Reply:
We can't just reset the batchID here, because the batchID of the same txn needs to be increased linearly. 
In addition, resetBatch may be called multiple times for a single row change event, so it needs to be idempotent.